src/Widgets/NavigationWidget/SubComponents/LanguageSwitch.tsx

```diff
@@ -0,0 +1,89 @@
+import { NavDropdown } from 'react-bootstrap'
+import {
+  ImageTag,
+  InPlaceEditingOff,
+  LinkTag,
+  Obj,
+  connect,
+  currentPage,
+  currentSiteId,
+} from 'scrivito'
+import { HomepageInstance } from '../../../Objs/Homepage/HomepageObjClass'
+
+export const LanguageSwitch = connect(function LanguageSwitch() {
+  const versions = Obj.onAllSites()
+    .where('_path', 'equals', '/')
+    .toArray()
+    .map((site) => {
+      const siteId = site.siteId()
+      const pageVersion = siteId && currentPage()?.versionOnSite(siteId)
+      const language = site.language() || 'en'
+      return {
+        label: displayName(language),
+        language,
+        version: pageVersion,
+        root: site as HomepageInstance,
+      }
+    })
+    .sort((a, b) => a.label.localeCompare(b.label))
+
+  if (versions.length < 2) return null
+
+  const activeSite = (
+    versions.find(({ root }) => root.siteId() === currentSiteId()) ||
+    versions[0]
+  )?.root
+
+  if (!activeSite) return null
+
+  const title = <LanguageLabel root={activeSite} showIconOnly />
```

Comment:
I would prefer, if the `showIconOnly` option gets removed and an `ImageTag` is used instead. It has the drawback, that `attribute="siteLanguageIcon"` and `className="img-flag"` are duplicated. But the code-flow is probably easier to understand for readers.

Reply:
If you look at the code after this refactoring we have quite some duplication. I tend to unify CurrentLanguageLabel and LanguageLabel again, adding a prop like `labelClassName`. WDYT?

Reply:
Sounds good to me.